service/src/main/java/uk/gov/hmcts/reform/fpl/service/DraftCMOService.java

```diff
@@ -42,18 +48,93 @@ public DraftCMOService(DateFormatterService dateFormatterService, ObjectMapper m
         this.directionHelperService = directionHelperService;
     }
 
-    public DynamicList getHearingDateDynamicList(CaseDetails caseDetails) {
-        Map<String, Object> data = caseDetails.getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+    public Map<String, Object> extractIndividualCaseManagementOrderObjects(
+        CaseManagementOrder caseManagementOrder,
+        List<Element<HearingBooking>> hearingDetails) {
 
-        List<Element<HearingBooking>> hearingDetails = caseData.getHearingDetails();
+        if (isNull(caseManagementOrder)) {
+            caseManagementOrder = CaseManagementOrder.builder().build();
+        }
+
+        HashMap<String, Object> data = new HashMap<>();
+        HashMap<String, Object> reviewCaseManagementOrder = new HashMap<>();
+
+        reviewCaseManagementOrder.put("cmoStatus", caseManagementOrder.getCmoStatus());
+
+        data.put("cmoHearingDateList", getHearingDateDynamicList(hearingDetails, caseManagementOrder));
+        data.put("schedule", caseManagementOrder.getSchedule());
+        data.put("recitals", caseManagementOrder.getRecitals());
+        data.put("reviewCaseManagementOrder", reviewCaseManagementOrder);
+
+        return data;
+    }
+
+    public CaseManagementOrder prepareCMO(Map<String, Object> caseData) {
```

Comment:
Method name is quite vague. As a reader who just starts with the project I would not know what that method does and why it is required. I'd expect to be read from case data in 'as is' way without any preparation.

Maybe Javadoc would shed some light?

Reply:
How about `buildCMOFromCaseData(Map?)`?